modules/nextflow/src/main/groovy/nextflow/container/SingularityCache.groovy

```diff
@@ -171,12 +173,23 @@ class SingularityCache {
     Path downloadSingularityImage(String imageUrl) {
         final localPath = localImagePath(imageUrl)
         final file = new File("${localPath.parent}/.${localPath.name}.lock")
+
         final wait = "Another Nextflow instance is pulling the Singularity image $imageUrl -- please wait the download completes"
         final err =  "Unable to acquire exclusive lock after $pullTimeout on file: $file"
 
+        if( localPath.exists() ) {
+            log.debug "Singularity found local store for image=$imageUrl; path=$localPath"
+            return localPath
+        }
+
+        final pullFile = Files.createTempFile(localPath.parent,
+                                              ".${localPath.name}".toString(),
+                                              ".pulling",
+                                              new FileAttribute<?>[0])
```

Comment:
What's the befits of using a temp file name for during the image pull if there's already a lock protecting against race conditions on the file download?